apps/invoices/src/modules/app-configuration/app-configuration-router.ts

```diff
@@ -0,0 +1,115 @@
+import { createLogger } from "@saleor/apps-shared";
+import { z } from "zod";
+import { protectedClientProcedure } from "../trpc/protected-client-procedure";
+import { router } from "../trpc/trpc-server";
+import { createSettingsManager } from "./metadata-manager";
+import { AppConfigV2MetadataManager } from "./schema-v2/app-config-v2-metadata-manager";
+import { GetAppConfigurationV2Service } from "./schema-v2/get-app-configuration.v2.service";
+import { PrivateMetadataAppConfiguratorV1 } from "./schema-v1/app-configurator";
+import { AppConfigV2 } from "./schema-v2/app-config";
+import { ConfigV1ToV2Transformer } from "./schema-v2/config-v1-to-v2-transformer";
+import { AppConfigV1 } from "./schema-v1/app-config-v1";
+import { Client } from "urql";
+import { ConfigV1ToV2MigrationService } from "./schema-v2/config-v1-to-v2-migration.service";
+import { trpcClient } from "../trpc/trpc-client";
+
+// todo unify
+const upsertAddressSchema = z.object({
+  address: z.object({
```

Comment:
If you extract the address to a separate schema, you should be able to use it with the form.

Reply:
extracted and unified